core/src/tools/bsmtp.cc

```diff
@@ -240,112 +208,87 @@ static void GetDateString(char* buf, int buf_len)
-        break;
-
-      case '6':
-        default_resolv_type = RESOLV_PROTO_IPV6;
-        break;
-
-      case '8':
-        content_utf8 = true;
-        break;
-
-      case 'a':
-        default_resolv_type = RESOLV_PROTO_ANY;
-        break;
-
-      case 'c':
-        Dmsg1(20, "cc=%s\n", optarg);
-        cc_addr = optarg;
-        break;
-
-      case 'd': /* set debug level */
-        if (*optarg == 't') {
-          dbg_timestamp = true;
-        } else {
-          debug_level = atoi(optarg);
-          if (debug_level <= 0) { debug_level = 1; }
-        }
-        Dmsg1(20, "Debug level = %d\n", debug_level);
-        break;
-
-      case 'f': /* from */
-        from_addr = optarg;
-        break;
-
-      case 'h': /* smtp host */
-        Dmsg1(20, "host=%s\n", optarg);
-        p = strchr(optarg, ':');
-        if (p) {
-          *p++ = 0;
-          mailport = atoi(p);
-        }
-        mailhost = optarg;
-        break;
-
-      case 's': /* subject */
-        Dmsg1(20, "subject=%s\n", optarg);
-        subject = optarg;
-        break;
-
-      case 'r': /* reply address */
-        reply_addr = optarg;
-        break;
-
-      case 'l':
-        Dmsg1(20, "maxlines=%s\n", optarg);
-        maxlines = (unsigned long)atol(optarg);
-        break;
-
-      case '?':
-      default:
-        usage();
-    }
-  }
-  argc -= optind;
-  argv += optind;
 
-  if (argc < 1) {
-    Pmsg0(0, _("Fatal error: no recipient given.\n"));
-    usage();
-    exit(1);
-  }
+  CLI::App bsmtp_app;
+  InitCLIApp(bsmtp_app, "The Bareos simple mail transport program.");
+  bsmtp_app.set_help_flag("--help,-?", "Print this help message and exit.");
+
+  resolv_type default_resolv_type = RESOLV_PROTO_IPV4;
+
+  bsmtp_app.add_flag(
+      "-4,--ipv4-protocol",
+      [&default_resolv_type](bool) { default_resolv_type = RESOLV_PROTO_IPV4; },
+      "Forces bsmtp to use IPv4 addresses only.");
+
+  bsmtp_app.add_flag(
+      "-6,--ipv6-protocol",
+      [&default_resolv_type](bool) { default_resolv_type = RESOLV_PROTO_IPV6; },
+      "Forces bsmtp to use IPv6 addresses only.");
+
+
+  bool content_utf8 = false;
+  bsmtp_app.add_flag("-8,--utf8", content_utf8, "set charset to UTF-8.");
+
+  bsmtp_app.add_flag(
+      "-a,--any-protocol",
+      [&default_resolv_type](bool) { default_resolv_type = RESOLV_PROTO_ANY; },
+      "Use any ip protocol for address resolution.");
+
+
+  std::string cc_addr{};
+  bsmtp_app.add_option("-c,--cc", cc_addr, "Set the Cc: field.");
+
+  AddDebugOptions(bsmtp_app);
+
+  std::string from_addr{};
+  bsmtp_app.add_option("-f,--from-address", from_addr, "Set the From: field.")
+      ->required();
+
+  int mailport = 25;
+  std::string mailhost{};
+  bsmtp_app
+      .add_option(
+          "-h,--mailhost",
+          [&mailport, &mailhost](std::vector<std::string> val) {
+            Dmsg1(20, "host=%s\n", val[0].c_str());
+            BStringList host_and_port(val[0], ':');
```

Comment:
This will not work with ipv6 address like 2001:DB8::baba:25 see general review comment for more details.

Reply:
We really want to allow people to give an ipv6 address for the mailhost, and the normal way to do that is to allow the form [2001:DB8::]:25
Would it be hard to implement that?